Tasks/SonarQubePostTest/SonarQubeBuildBreaker.ps1

```diff
@@ -13,7 +13,8 @@ function BreakBuildOnQualityGateFailure
             Write-Host "Ignoring the setting of breaking the build on quality gate failure because the build was triggered by a pull request."
             return;
         }
-       
+        
+        WaitForAnalysisToFinish
```

Comment:
As advised by Duncan, I am making the call to wait for the analysis to finish explicitly instead of having it inside GetOrFetchQualityGateStatus